### PR TITLE
DD-39 corrected styles and fixed bags

### PR DESCRIPTION
**Pull request by YuraStork**

### CODE REVIEW

File: src/pages/home/activeRooms/index.tsx [hunk 1]
```diff
@@ -1,27 +1,41 @@
...
 
 type ActiveRoomsProps = {
   activeRooms: ActiveRoom[]
   userId: string
 }
 
 export const ActiveRooms: FC<ActiveRoomsProps> = ({ activeRooms, userId }) => {
+  const sortedActiveRooms = useMemo(() => {
+    const str = JSON.stringify(activeRooms)
+    const arr = jsonParse<ActiveRoom[]>(str)
```
Comment (Andrew01011997) [inline src/pages/home/activeRooms/index.tsx]:
```suggestion
    const arrCopied = jsonParse<ActiveRoom[]>(JSON.stringify(activeRooms))
```

File: src/pages/home/activeRooms/index.tsx [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
+
+    return arr?.sort((a) => (a.owner === userId ? -1 : 0))
+  }, [activeRooms])
+
+  if (!sortedActiveRooms) return <p>no active rooms</p>
```
Comment (Andrew01011997) [inline src/pages/home/activeRooms/index.tsx]: Should you also check the `length`?

File: src/pages/home/activeRooms/index.tsx [hunk 3]
```diff
@@ -1,27 +1,41 @@
...
+      {sortedActiveRooms.map((room) => (
+        <ActiveRoomWrapper key={room._id} isCurrentUserRoom={room.owner === userId}>
+          <ColorCircle color={colors.green} />
+          <Heading4>{room.roomName}</Heading4>
+          <Heading5>
+            {room.users.length} | {room.limit}
+          </Heading5>
+        </ActiveRoomWrapper>
+      ))}
     </>
```
Comment (Andrew01011997) [inline src/pages/home/activeRooms/index.tsx]: coudl you remove fragment and `{` from here?

File: src/pages/home/activeRooms/styles.tsx [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
+    background-color: ${({ theme }) => theme.colors.darkCyan};
+    box-shadow: 0px 0px 3px 1px ${({ theme }) => theme.colors.whiteSmoke};
   }
-`
 
-export { Room }
+  & > h4 {
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
```
Comment (Andrew01011997) [inline src/pages/home/activeRooms/styles.tsx]: you can create a `css` for this and extend it here

File: src/pages/home/createRoom/index.tsx [hunk 5]
```diff
@@ -28,6 +30,8 @@ export const CreateRoomComponent: FC<ComponentProps> = ({ isLoading, setIsLoadin
   const formik = useFormik({
     initialValues,
     validationSchema,
+    validateOnChange: false,
+    validateOnBlur: false,
```
Comment (Andrew01011997) [inline src/pages/home/createRoom/index.tsx:34]: in this case validation will only on `submit`?

File: src/pages/home/enterInRoom/index.tsx [hunk 6]
```diff
@@ -28,43 +30,38 @@ export const EnterInRoomComponent: FC<EnterInRoomProps> = ({ isLoading, setIsLoa
...
-        </div>
-        <SubmitButton type='submit' disabled={isLoading || !formik.dirty || !formik.isValid}>
-          Enter in room
-        </SubmitButton>
-      </form>
-    </RoomWrapper>
+
+          <SubmitButton type='submit' disabled={isLoading || (!formik.dirty && !formik.isValid)}>
+            Enter in room
+          </SubmitButton>
```
Comment (Andrew01011997) [inline src/pages/home/enterInRoom/index.tsx]: On my opinion `SubmitButton` should contains `type="submit"` inside

**Review by Andrew01011997 — CHANGES_REQUESTED** [review]

File: src/pages/home/createRoom/index.tsx (hunk 5 above)
Comment (YuraStork) [reply src/pages/home/createRoom/index.tsx:34]: yes it so

**Review by Andrew01011997 — DISMISSED** [review]

**Review by Andrew01011997 — APPROVED** [review]

**Pull request merged**